src/App.tsx

```diff
@@ -1,61 +1,159 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser, User } from './react-app-env';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (userId: number): User | null => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const todoWithUser: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
+  const [todos, setTodos] = useState<TodoWithUser[]>(todoWithUser);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const addTodo = () => {
+    const maxId = Math.max(...todos.map(todo => todo.id));
+    const newTodo = {
+      id: maxId + 1,
+      title,
+      completed: false,
+      userId: selectedUserId,
+      user: usersFromServer.find(user => user.id === selectedUserId) || null,
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const clearButtonClick = () => {
+    setIsButtonClicked(false);
+  };
+
+  const clearTitle = () => {
+    setTitle('');
+  };
+
+  const clearSelectedUserId = () => {
+    setSelectedUserId(0);
+  };
```

Comment:
You can combine the last two methods into a single reset method and move the button click logic right into the handle method.

Reply:
Yes, but I tried to follow this recommendation^_^
![image](https://user-images.githubusercontent.com/110456763/198298546-48f548e9-f8fe-4882-abb8-c3bed5588a9c.png)